plugins/out_pgsql/pgsql.c

```diff
@@ -0,0 +1,252 @@
+    if(!ctx->db_user) {
+        flb_warn("[out_pgsql] You didn't supply a valid user to connect, your current unix user will be used");
+    }
+
+    ctx->db_passwd = flb_output_get_property("password", ins);
+    if(!ctx->db_passwd) {
+        flb_warn("[out_pgsql] You didn't supply a password, you should use a password to authenticate against PostgreSQL");
+    }
+
+    ctx->timestamp_key = flb_sds_create(flb_output_get_property("timestamp_key", ins));
+    if(!ctx->timestamp_key) {
+        ctx->timestamp_key = flb_sds_create(FLB_PGSQL_TIMESTAMP_KEY);
+        if(ctx->timestamp_key == NULL) {
+            pgsql_conf_destroy(ctx);
+            return -1;
+        }
+    }
+
+    flb_info("[out_pgsql] host=%s port=%s dbname=%s ...",
+             ctx->db_hostname, ctx->db_port, ctx->db_name);
+
+    ctx->conn = PQsetdbLogin(ctx->db_hostname,
+                             ctx->db_port,
+                             NULL, NULL,
+                             ctx->db_name,
+                             ctx->db_user,
+                             ctx->db_passwd);
+
+    if(PQstatus(ctx->conn) != CONNECTION_OK) {
+        flb_error("[out_pgsql] failed to connect to host=%s with error: %s",
+                  ctx->db_hostname, PQerrorMessage(ctx->conn));
+
+        pgsql_conf_destroy(ctx);
+
+        return -1;
+    }
+
+    flb_info("[out_pgsql] host=%s port=%s dbname=%s OK",
+              ctx->db_hostname, ctx->db_port, ctx->db_name);
+    flb_output_set_context(ins, ctx);
+
+    temp = PQescapeIdentifier(ctx->conn, ctx->db_table, flb_sds_len(ctx->db_table));
+    flb_sds_destroy(ctx->db_table);
+    ctx->db_table = flb_strdup(temp);
+    flb_free(temp);
+
+    flb_info("[out_pgsql] we check that the table %s exists, if not we create it", ctx->db_table);
+
+    str_len = 62 + flb_sds_len(ctx->db_table);
+
+    query = flb_malloc(str_len);
+    if(query == NULL) {
+        pgsql_conf_destroy(ctx);
+        return -1;
+    }
+
+    snprintf(query, str_len, "CREATE TABLE IF NOT EXISTS %s (data json);", ctx->db_table);
+    res = PQexec(ctx->conn, query);
+    flb_free(query);
+
+    if(PQresultStatus(res) != PGRES_COMMAND_OK) {
+        flb_error("[out_pgsql] %s", PQerrorMessage(ctx->conn));
+        pgsql_conf_destroy(ctx);
+        return -1;
+    }
+
+    PQclear(res);
+
+    flb_info("[out_pgsql] switching postgresql connection to non-blocking mode");
+    if(PQsetnonblocking(ctx->conn, 1) != 0) {
+        flb_error("[out_pgsql] non-blocking mode not set, not good");
+        pgsql_conf_destroy(ctx);
+        return -1;
+    }
+
+    return 0;
+}
+
+static void cb_pgsql_flush(const void *data, size_t bytes,
+                            const char *tag, int tag_len,
+                            struct flb_input_instance *i_ins,
+                            void *out_context,
+                            struct flb_config *config)
+{
+    struct flb_pgsql_config *ctx = out_context;
+    flb_sds_t json;
+    PGresult *res = NULL;
+    int result;
```

Comment:
When I build this, I get:

```
/home/ec2-user/fluent-bit/plugins/out_pgsql/pgsql.c:175:9: warning: variable ‘result’ set but not used [-Wunused-but-set-variable]
     int result;
         ^~~~~~
```

Reply:
Yep and the variable it' used in line 215, we don't use the value because we check the result in line 222, maybe we can just drop it, what do you think ?

Reply:
I read this in the postgre library doc:

> 1 is returned if the command was successfully dispatched and 0 if not (in which case, use PQerrorMessage to get more information about the failure).

> maybe we can just drop it, what do you think ?

I don't know enough about the postgre library to say; you decide. If you prefer the check on line 222 and it covers all cases such that a check here is not needed, then I'd say drop the `result` var.

Reply:
Yeah better drop for now since we do the PQflush() donw and that will dispatch the content if PQsendQuery() didn't, anyway in the future we need to improve over the async task if possible but we may need to define a base version of PostgreSQL to support